minio/minio
# Added attribute proxy for `mc admin config set ALIAS logger_webhook`

## Description
Added an additional command line option to set proxy for loger_wehook option in `mc admin config set` command.

## Motivation and Context
We should be able to set proxy for logger webhook and use the same while pushing the logs from minio to external wehooks

## How to test this PR?
Follow the below steps to test the PR
Step-1: Setup a generic webservice for logs (e.g. https://webhook.site/). Copy the `Unique URL`

![Screenshot from 2023-02-20 16-02-40](https://user-images.githubusercontent.com/6771252/220081022-4bb586fc-8b01-4083-b736-e1917af600d9.png)

Step-2: Start the minio instance
```
✘ ./minio server ./data/1p1host{1...4}/dist{1...4} --address :9010
```
Step-3: Create the `mc alias` for minio instance
```
✘ ./mc alias set myminio http://localhost:9010 minioadmin minioadmin
```
Step-4: Setup a local proxy server (squid) and start it. By default it runs at port 3128
```
✘ sudo systemctl start squid
```
Step-5: Check if proxy server working
```
✘ curl -x http://localhost:3128 -I http://google.com
HTTP/1.1 301 Moved Permanently
Location: http://www.google.com/
Content-Type: text/html; charset=UTF-8
Date: Mon, 20 Feb 2023 03:28:37 GMT
Expires: Wed, 22 Mar 2023 03:28:37 GMT
Cache-Control: public, max-age=2592000
Server: gws
Content-Length: 219
X-XSS-Protection: 0
X-Frame-Options: SAMEORIGIN
X-Cache: MISS from fedora-shubhendu
X-Cache-Lookup: MISS from fedora-shubhendu:3128
Via: 1.1 fedora-shubhendu (squid/5.7)
Connection: keep-alive
```
Step-6: Create a bucket and copy an object to minio instance
```
✘ ./mc ls test-bucket myminio
✘ ./mc cp /etc/hosts myminio/test-bucket
```

Step-7: Set the logger webhook
```
✘ ./mc admin config set myminio logger_webhook enable=on endpoint=<Unique URL from Step-1>
```
Step-8: Create a problem with minio object so that it can create some logs
```
✘ cd ./data/
✘ chmod 000 1p1host1 <--- removing the permissions
```
Step-9: List the bucket content
```
✘ ./mc ls myminio/test-bucket
[2023-02-17 17:54:52 IST]   633B STANDARD hosts
```
Step-10: Check the webhook.site logs

![Screenshot from 2023-02-20 16-05-40](https://user-images.githubusercontent.com/6771252/220081688-57cee05d-d788-4b4c-b15c-096736cecfb5.png)

Step-11: Now enable and set the proxy for logger webhook
```
✘ ./mc admin config set myminio logger_webhook proxy=http://localhost:3128
Successfully applied new settings.
✘ ./mc admin config get myminio logger_webhook
logger_webhook endpoint=https://webhook.site/ee3c44fc-3b2d-494c-9a95-5b79ce1f58b2 auth_token= client_cert= client_key= proxy=http://localhost:3128 queue_size=100000
```
Step-12: Again, run the listing of bucket and you should see the log as below
```
✘ ./mc ls myminio/test-bucket
[2023-02-17 17:54:52 IST]   633B STANDARD hosts
```

![Screenshot from 2023-02-20 16-16-05](https://user-images.githubusercontent.com/6771252/220083938-470a3bee-548e-409b-8feb-302b7862a288.png)

Step-13: Now set an invalid proxy value for logger webhook as below
```
✘ ./mc admin config set myminio logger_webhook proxy=http://invalid.proxy.com
Successfully applied new settings.
✘ ./mc admin config get myminio logger_webhook
logger_webhook endpoint=https://webhook.site/ee3c44fc-3b2d-494c-9a95-5b79ce1f58b2 auth_token= client_cert= client_key= proxy=http://invalid.proxy.com queue_size=100000 
```
This should cause an error as below in minio console logs
```
API: SYSTEM()
Time: 10:48:13 UTC 02/20/2023
DeploymentID: e33b0b45-e564-43ab-b03a-604e27489a09
Error: Unable to update logger webhook config: [Post "https://webhook.site/ee3c44fc-3b2d-494c-9a95-5b79ce1f58b2": proxyconnect tcp: lookup invalid.proxy.com on 127.0.0.53:53: no such host] (*errors.errorString)
       5: internal/logger/logger.go:258:logger.LogIf()
       4: cmd/config-current.go:576:cmd.applyDynamicConfigForSubSys()
       3: cmd/admin-handlers-config-kv.go:111:cmd.applyDynamic()
       2: cmd/admin-handlers-config-kv.go:177:cmd.adminAPIHandlers.SetConfigKVHandler()
       1: net/http/server.go:2109:http.HandlerFunc.ServeHTTP()
```
Step-14: Now try listing the bucket and logs wont reach webhook.site
```
✘ ./mc ls myminio/test-bucket
[2023-02-17 17:54:52 IST]   633B STANDARD hosts
```
The error logs at minio console
```
API: SYSTEM()
Time: 10:53:48 UTC 02/20/2023
DeploymentID: e33b0b45-e564-43ab-b03a-604e27489a09
Error: open /home/shubhendu/work-forked-repos/minio/data/1p1host1/dist1/.minio.sys/buckets/.tracker.bin: permission denied (*fs.PathError)
       2: internal/logger/logger.go:258:logger.LogIf()
       1: cmd/data-update-tracker.go:303:cmd.(*dataUpdateTracker).startSaver()

API: SYSTEM()
Time: 10:53:48 UTC 02/20/2023
DeploymentID: e33b0b45-e564-43ab-b03a-604e27489a09
Error: open /home/shubhendu/work-forked-repos/minio/data/1p1host1/dist2/.minio.sys/buckets/.tracker.bin: permission denied (*fs.PathError)
       2: internal/logger/logger.go:258:logger.LogIf()
       1: cmd/data-update-tracker.go:303:cmd.(*dataUpdateTracker).startSaver()

API: SYSTEM()
Time: 10:53:48 UTC 02/20/2023
DeploymentID: e33b0b45-e564-43ab-b03a-604e27489a09
Error: open /home/shubhendu/work-forked-repos/minio/data/1p1host1/dist3/.minio.sys/buckets/.tracker.bin: permission denied (*fs.PathError)
       2: internal/logger/logger.go:258:logger.LogIf()
       1: cmd/data-update-tracker.go:303:cmd.(*dataUpdateTracker).startSaver()

API: SYSTEM()
Time: 10:53:48 UTC 02/20/2023
DeploymentID: e33b0b45-e564-43ab-b03a-604e27489a09
Error: open /home/shubhendu/work-forked-repos/minio/data/1p1host1/dist4/.minio.sys/buckets/.tracker.bin: permission denied (*fs.PathError)
       2: internal/logger/logger.go:258:logger.LogIf()
       1: cmd/data-update-tracker.go:303:cmd.(*dataUpdateTracker).startSaver()
```

But no new logs on webhook.site. Only older logs present above

![Screenshot from 2023-02-20 16-25-35](https://user-images.githubusercontent.com/6771252/220085922-8e36da9f-7edb-454b-87ff-93b575066a4d.png)

## Types of changes

## Checklist:
